solidity-checks/src/main/java/org/sonarsource/solidity/checks/InvertedBooleanCheck.java

```diff
@@ -0,0 +1,22 @@
+package org.sonarsource.solidity.checks;
+
+import org.antlr.v4.runtime.tree.ParseTree;
+import org.sonar.check.Rule;
+import org.sonarsource.solidity.frontend.SolidityParser.ExpressionContext;
+
+@Rule(key = InvertedBooleanCheck.RULE_KEY)
+public class InvertedBooleanCheck extends IssuableVisitor {
+
+  public static final String RULE_KEY = "ExternalRule5";
+
+  @Override
+  public ParseTree visitExpression(ExpressionContext ctx) {
+    ParseTree expr = ctx.getChild(0);
+    if (ctx.primaryExpression() == null && expr.getText().startsWith("!")) {
+      if (CheckUtils.isParenthesized(ctx.getChild(1)) && CheckUtils.isBooleanExpression(CheckUtils.removeParenthesis(ctx.getChild(1)))) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Merge this if statement with the enclosing one. ([squid:S1066](https://sonarcloud.io/organizations/stylianos-agapiou-sonarsource-gi/rules#rule_key=squid%3AS1066))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=org.sonarsource.solidity%3Asonar-solidity&pullRequest=35&issues=AWR-TYh0I66nRh8YLRI6&open=AWR-TYh0I66nRh8YLRI6)

[ISSUE_KEY]: AWR-TYh0I66nRh8YLRI6